csm/cli/support_bundle/csm_bundle_generate.py

```diff
@@ -36,14 +41,41 @@ async def init(command):
         :return:
         """
         # Read Config to Fetch Log File Path
-        log_directory_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log.log_path")
+        csm_log_directory_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log.log_path")
+        alerts_file_path = Conf.get(const.CSM_GLOBAL_INDEX, "SUPPORT_BUNDLE.alerts_file_path")
+        # Fetch alerts for support bundle.
+        alerts_data = await CSMBundle.fetch_and_save_alerts()
+        obj_alert_json = Json(alerts_file_path)
+        obj_alert_json.dump(alerts_data)
         # Creates CSM Directory
         path = command.options.get("path")
         bundle_id = command.options.get("bundle_id")
-        path = os.path.join(path, const.CSM_GLOBAL_INDEX)
-        os.makedirs(path)
+        component_name = command.options.get("component", "csm")
+        component_data = {"csm": [csm_log_directory_path],
+                          "alerts": [alerts_file_path]}
+        temp_path = os.path.join(path, component_name)
+        os.makedirs(temp_path, exist_ok = True)
         # Generate Tar file for Logs Folder.
-        tar_file_name = os.path.join(path, f"csm_{bundle_id}.tar.gz")
-        Tar(tar_file_name).dump([log_directory_path])
-
+        tar_file_name = os.path.join(path, f"{component_name}_{bundle_id}.tar.gz")
+        if all(map(os.path.exists, component_data[component_name])):
+            Tar(tar_file_name).dump(component_data[component_name])
+        else:
+            raise CsmError(rc = errno.ENOENT,
+                           desc = f"Component log missing: {component_data[component_name]}")
 
+    @staticmethod
+    async def fetch_and_save_alerts():
+        """
+        Fetches the alerts from es db and creates a json file
+        :param command: Csm_cli Command Object :type: command
+        :return: None
+        """
+        alerts =[]
+        try:
+            conf = GeneralConfig(Yaml(const.DATABASE_CONF).load())
+            db = DataBaseProvider(conf)
+            repo = AlertRepository(db)
+            alerts = await repo.fetch_alert_for_support_bundle()
+        except Exception as ex:
+            Log.exception(f"Error occured while fetching alerts: {ex}")
```

Comment:
In case of exception, log the error in alerts.json - Internal error : could not fetch alerts.

Reply:
Done